src/AspNet.Security.OAuth.Apple/AppleAuthenticationOptions.cs

```diff
@@ -0,0 +1,35 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.AspNetCore.Authentication.OAuth;
+
+namespace AspNet.Security.OAuth.Apple
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="AppleAuthenticationHandler"/>.
+    /// </summary>
+    public class AppleAuthenticationOptions : OAuthOptions
```

Comment:
According to the [docs](https://developer.apple.com/documentation/signinwithapplerestapi/generate_and_validate_tokens#3262048), I think it should have an option field for a private key, so that the provider can generate the credentials programmatically.

Reply:
Yeah, I thought about that, but I’ll need to sort out a working setup in Apple first so I’ve got something to play around with to test it with 🙂

Reply:
This is now possible by setting `GenerateClientSecret` to `true` and then either returning the private key's bytes via the `Func<string, Task<byte[]>> PrivateKeyBytes` property or providing a custom implementation of the `AppleKeyStore` class.